### PR TITLE
Get the version number using git tags

### PR DESCRIPTION
**Pull request by nbelakovski**
This lets us avoid having a hardcoded file in the repo and having to update it for a release. This way releases can be done simply by tagging the appropriate commit and rebuilding.

This will be useful for when we publish packages, since it lets us publish a package every time we run CI and not have it conflict with or overwrite the latest package.

### CODE REVIEW

**Comment by jschueller** [comment]
this will fail on windows or if e build from the github tarball

**Comment by nbelakovski** [comment]
It works on Windows, I specifically modified the code so that it would. Why would we build from the github tarball?

**Comment by zaikunzhang** [comment]
@nbelakovski, thank you for proposing this. Ping me when both you and @jschueller think it is ready.

**Comment by jschueller** [comment]
it wont work on windows without bash support, why not call directly the git command from cmake instead ?

**Comment by jschueller** [comment]
also it should work from the tarball, macports already uses it, and so will linux distro usually

**Comment by nbelakovski** [comment]
I changed it to use `git describe` so that it's not necessary to use a bash script. I had actually initially wanted to use it, but I think I didn't like the default output and so I wanted to make my own but long story short I'd rather keep things simple.

I also learned about `export-subst` in this process, which lets you write a file in such a way that `git archive` will substitute certain placeholders in the file with git information, and can even include the output of describe. This should solve the macports case.

I validated that the .git-archival.txt file is set up correctly by performing the following actions on my machine (I obtained the githash from going into the actions and looking at the output of the clone repository action):
```bash
$ curl -L https://api.github.com/repos/libprima/prima/tarball/9a0485ea -o myarchive.tgz
$ tar xzf myarchive.tgz
$ cd libprima-prima-9a0485e
$ cat .git-archival.txt
v0.7.2-162-g9a0485ea5
```

Note that the above commands will return a blank output on forked repos since they do not copy tags by default.

File: CMakeLists.txt [hunk 1]
```diff
@@ -62,7 +62,29 @@ if (PRIMA_ENABLE_C)
...
+  file(STRINGS .git-archival.txt PRIMA_VERSION)
+  if(PRIMA_VERSION MATCHES "describe")
+    message(WARNING "No git detected and .git-archival.txt does not contain a version number")
+    set(PRIMA_VERSION "unknown")
+  endif()
+  
+endif()
+# Remove the leading v from PRIMA_VERSION, if it contains one.
+string(REGEX REPLACE "^v" "" PRIMA_VERSION ${PRIMA_VERSION})
+message("Setting PRIMA version to ${PRIMA_VERSION}")
```
Comment (jschueller) [inline CMakeLists.txt:87]: use `message(STATUS ...)`

**Comment by zaikunzhang** [comment]
Hi @nbelakovski and @jschueller , is the PR ready? Thank you.

**Review by jschueller — APPROVED** [review]

**Pull request merged**